docs/existing_data.rst

```diff
@@ -10,18 +10,51 @@ To deploy a Patroni cluster without using a pre-existing PostgreSQL instance, se
+    #. Create the Postgres users as defined in the :ref:`authentication <postgresql_settings>` section of the Patroni configuration. You can find sample SQL commands to create the users in the code block below. Replace the usernames and passwords as per your configuration.
 
-   CREATE USER $PATRONI_SUPERUSER_USERNAME WITH SUPERUSER ENCRYPTED PASSWORD '$PATRONI_SUPERUSER_PASSWORD';
-   CREATE USER $PATRONI_REPLICATION_USERNAME WITH REPLICATION ENCRYPTED PASSWORD '$PATRONI_REPLICATION_PASSWORD';
+        .. code-block:: sql
 
-3. Start Patroni (e.g. ``patroni /etc/patroni/patroni.yml``). It automatically detects that PostgreSQL daemon is already running but its configuration might be out-of-date.
-4. Ask Patroni to restart the node with ``patronictl restart cluster-name node-name``. This step is only required if PostgreSQL configuration is out-of-date.
+           -- Patroni superuser
+           CREATE USER PATRONI_SUPERUSER_USERNAME WITH SUPERUSER ENCRYPTED PASSWORD '$PATRONI_SUPERUSER_PASSWORD';
+
+           -- Patroni replication user
+           CREATE USER PATRONI_REPLICATION_USERNAME WITH REPLICATION ENCRYPTED PASSWORD '$PATRONI_REPLICATION_PASSWORD';
+
+           -- Patroni rewind user, if you enabled use_pg_rewind in your configuration
+           CREATE USER PATRONI_REWIND_USERNAME WITH ENCRYPTED PASSWORD '$PATRONI_REWIND_PASSWORD';
+           GRANT EXECUTE ON function pg_catalog.pg_ls_dir(text, boolean, boolean) TO PATRONI_REWIND_USERNAME;
+           GRANT EXECUTE ON function pg_catalog.pg_stat_file(text, boolean) TO PATRONI_REWIND_USERNAME;
+           GRANT EXECUTE ON function pg_catalog.pg_read_binary_file(text) TO PATRONI_REWIND_USERNAME;
+           GRANT EXECUTE ON function pg_catalog.pg_read_binary_file(text, bigint, bigint, boolean) TO PATRONI_REWIND_USERNAME;
+
+    #. Start Patroni using the `patroni` systemd service unit. It automatically detects that Postgres is already running and starts monitoring the instance.
+
+#. Hand over Postgres "start up procedure" to Patroni. In order to do that you need to restart the cluster members through ``patronictl restart cluster-name member-name`` command. For minimal downtime you might want to split this step into:
+
+    #. Immediate restart of the standby nodes.
+    #. Scheduled restart of the primary node within a maintenance window.
```

Comment:
1. Restart order depends on mismatch of actual postgres configuration and desired. For example if postgres runs with max_connections = 1000 and you set it to 500, the procedure will not work. I.e., Patroni will start replica back with the original value (1000).
2. Patroni might automatically restart replica nodes in case if primary_conninfo, primary_slot_name, restore_command or similar doesn't match, but only for Postgres older than 13.

Reply:
My idea for this documentation was that users would migrate an existing cluster and *do not* change configuration.
Maybe we could a put an explicit note about that assumption in the docs? What do you think?

BTW this PR https://github.com/zalando/patroni/pull/2786 could come handy here in regards to creating an YAML file, huh? :)